Add upper limit on DataStreams and WeakRefStrings version

nalimilan opened the pull request:
Next major releases of DataStreams and WeakRefStrings will move from `Nullable` to `Union{T, Null}` and will not work with the package in its current state.

Cf. https://github.com/JuliaLang/METADATA.jl/pull/10915.

cjprybol commented:
~@quinnj does this need an upper limit to WeakRefStrings too? Looks like the latest release `v0.3.0` of WRS includes the new changes for Nulls and DataTables is targeting the format used in `v0.2.0`~ yes, https://github.com/JuliaLang/METADATA.jl/pull/10915#issuecomment-324937607

quinnj commented:
The DataStreams upper bound should already include the WRS bound, so not technically, but couldn't hurt.

nalimilan committed:
Add upper limit on DataStreams and WeakRefStrings version

Next major releases of DataStreams and WeakRefStrings will move from
`Nullable` to `Union{T, Null}` and will not work with the package
in its current state.

nalimilan commented:
I've also added the upper bound for WeakRefStrings, can't hurt.

cjprybol commented:
Failing on nightlies due to an issue in CSV https://ci.appveyor.com/project/nalimilan/datatables-jl/build/1.0.379/job/ddcpq7vwdvg5dwdd#L4871. I've tagged a new release which should include the fix https://github.com/JuliaLang/METADATA.jl/pull/10923
